include/ipm/Receiver.hpp

```diff
@@ -77,8 +79,6 @@ class Receiver
 
   Receiver() = default;
 
-  virtual ~Receiver() = default;
```

Comment:
Don't we need a virtual dtor here so we can delete derived classes through a `Receiver*`?

Reply:
You're right...I had removed them because they weren't being used, but `https://en.cppreference.com/w/cpp/language/rule_of_three` would tend to require it be defined anyway.